app/controllers/concerns/pagination.rb

```diff
@@ -0,0 +1,42 @@
+module Pagination
+  extend ActiveSupport::Concern
+
+  included do
+    helper_method :per_page, :pagination_block, :current_page
+
+    private
+
+    def per_page
+      params[:per] || self.class.class_variable_get(:@@_pagination_per)
+    end
+
+    def pagination_block(model)
+      return [] if current_page.zero?
+      total_pages = calculate_total_pages(model.unscope(:offset, :limit).count)
+      pages = []
+      3.downto(1).each { |index| pages << current_page - index }
+      pages << current_page
+      1.upto(3).each { |index| pages << current_page + index }
+      pages.delete_if { |page| page < 1 || page > total_pages }
+      pages
+    end
+
+    def calculate_total_pages(count)
+      count / per_page + ((count % per_page.to_f).zero? ? 0 : 1)
+    end
+
+    def current_page
+      if @current_page.nil?
+        @current_page = params[:page].blank? ? 1 : params[:page].to_i
+      end
+
+      @current_page
+    end
+  end
+
+  class_methods do
+    def per_page_default(per)
+      class_variable_set(:@@_pagination_per, per)
```

Comment:
controller method for setting default per page items amount

Reply:
Why are you using a class variable?

How would this work in a multi-threaded situation?

Reply:
This variable needed only for default value amount items per page. I set it in controller and in multi-thread it will work properly.